linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -69,91 +89,103 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
 		}
 	}
 
-	// Ensure that the DistributionPoint is a reasonable URI. To get to the URI,
-	// we have to step inside the DistributionPointName, then step inside that's
-	// FullName, and finally read the singular SEQUENCE OF GeneralName element.
-	if !idpv.PeekASN1Tag(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) {
-		return &lint.LintResult{
-			Status:  lint.Warn,
-			Details: "IDP should contain distributionPoint",
+	if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(0).ContextSpecific().Constructed()) || idpv.PeekASN1Tag(cryptobyte_asn1.Tag(1).ContextSpecific()) {
```

Comment:
The peeking starts to become a little hard to reason about. Since we actually do care about these values, I think it would be cleaner to call [`ReadOptionalASN1`](https://pkg.go.dev/golang.org/x/crypto/cryptobyte#String.ReadOptionalASN1) three times, once for `distributionPoint [0]`, once for `onlyContainsUserCerts [1]`, and once for `onlyContainsCACerts [2]`. Then once we've done all the parsing we can check the constraints that we care about: these fields must obey one of:

 - onlyContainsUserCerts present and true, onlyContainsCACerts absent, distributionPoint contains a single, parseable, HTTP URL.
 - onlyContainsUserCerts absent, onlyContainsCACerts present and true, distributionPoint ??